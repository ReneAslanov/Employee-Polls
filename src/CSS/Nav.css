*{
    box-sizing: border-box;
    margin: 0;
    padding: 0;
}

.nav-wrapper{
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    border-bottom: 1px solid #e1e5eb;
}

.nav-list{
    padding: 1rem 2rem 2rem;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;

}

.nav-list a
{
    margin-right: 2em;
    text-decoration: none;
    color: #000;
    font-size: 1.2em;
    font-weight: 300;
    margin-top: 1em;
}

.nav-list a:hover{
    border-bottom: 1px solid #000;
}

.nav-avatar{
    height: 3.5rem;
    width: 3.5rem;
    position: absolute;
    top: 1.3em;
    margin: 0 0 0 -4.5em;
}

.nav-img-wrapper{
    display: flex;
}


.nav-list p
{
    margin-right: 2em;
    color: #b3b8c0;
    font-size: 1.2em;
    font-weight: 300;
    margin-top: 1em;
}