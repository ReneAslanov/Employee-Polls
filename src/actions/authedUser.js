export const SET_AUTHED_USER = "SET_AUTHED_USER";
export const LOGOUT = "LOGOUT";
export const SET_POLL_OPTION = "SET_POLL_OPTION";

export function setAuthedUser(payload)
{
    return{
        type: SET_AUTHED_USER,
        id: payload.id,
        avatar: payload.avatarURL,
        answers: payload.answers,
        questions: payload.questions
    }
}

export function logoutUser()
{
    return{
        type: LOGOUT
    }
}

export function setPollOption({qid, answer})
{
    return{
        type: SET_POLL_OPTION,
        qid,
        answer
    }
}