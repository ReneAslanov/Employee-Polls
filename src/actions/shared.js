export const RECEIVE_USERS = "RECEIVE_USERS";
export const SET_POLL_ANSWER = "SET_POLL_ANSWER";
export const SET_QUESTION_ARRAY = "SET_QUESTION_ARRAY";

export function setUserAnswer({authedUser, qid, answer})
{
    return{
        type: SET_POLL_ANSWER,
        authedUser,
        qid,
        answer
    }
}

export function setUserQuestionArray({authedUser, qid})
{
  return{
    type: SET_QUESTION_ARRAY,
    authedUser,
    qid
  }
}

export function receiveUsers(payload)
{
    return{
        type: RECEIVE_USERS,
        users: payload
    }
}
