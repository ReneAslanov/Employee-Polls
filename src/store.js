import { legacy_createStore as createStore} from "redux";
import middleware from './middleware';
import reducer from "./reducers/index";

const store = createStore(reducer, middleware);

export default store;

