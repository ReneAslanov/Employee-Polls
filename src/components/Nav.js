import { Link , useNavigate} from "react-router-dom";
import "../CSS/Nav.css";
import { connect } from "react-redux";

function Nav(props)
{
    const navigate = useNavigate();
    function logout()
    {
        navigate("/");
    }

    return(
        <div className="nav-wrapper">
            <div className="nav-routes">
                <ul className="nav-list">
                   <Link to="/home">Home</Link>
                   <Link to="/leaderboard">Leaderboard</Link>
                   <Link to="/add">New</Link>
                </ul>
            </div>
            <div className="nav-user">
                <ul className="nav-list">
                    <div className="nav-img-wrapper">
                        <img src={props.authedUser.avatar} alt="just an avatar" className="nav-avatar"/>
                    </div>
                    <p className="nav-userName">
                        {props.authedUser.id}
                    </p>
                    <Link to="/" onClick={logout}>Logout</Link>
                </ul>
            </div>
        </div>
    )
}

function mapStateToProps({authedUser})
{
    return{
        authedUser
    }
}

export default connect(mapStateToProps)(Nav);