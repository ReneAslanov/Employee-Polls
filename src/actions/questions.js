import { _saveQuestionAnswer } from "../utils/_DATA.js"
import { SET_POLL_OPTION } from "./authedUser.js";

export const SET_QUESTIONS = "SET_QUESTIONS";
export const CREATE_QUESTION = "CREATE_QUESTION";

export function setQuestions(questions)
{
    return{
        type: SET_QUESTIONS,
        questions : Object.values(questions)
    }
}

export function createQuestion(question)
{
    return{
        type: CREATE_QUESTION,
        payload: question
    }
}

export function handleCreateQuestion(optionOne, optionTwo, author)
{

    return{
        optionOneText: optionOne,
        optionTwoText: optionTwo,
        author
    }
}

export function handleQuestionAnswer({authedUser, qid, answer})
{
    return (dispatch) => {
        return _saveQuestionAnswer({
            authedUser,
            qid,
            answer
        })
        .then(data => {
            return dispatch(setPollAnswer({authedUser, qid, answer}))
        })

    }
}

export function setPollAnswer({qid, answer, authedUser})
{
    return{
        type: SET_POLL_OPTION,
        qid,
        answer,
        authedUser
    }
}