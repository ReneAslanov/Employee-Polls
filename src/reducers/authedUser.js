import { SET_AUTHED_USER, SET_POLL_OPTION } from "../actions/authedUser";
import { LOGOUT } from "../actions/authedUser";
import { CREATE_QUESTION } from "../actions/questions";

export default function authedUser(state = null, action)
{
    switch(action.type)
    {
        case SET_AUTHED_USER:
            return{
               id: action.id,
               avatar: action.avatar,
               answers: action.answers,
               questions: action.questions
            };

        case LOGOUT:
            return null

        case CREATE_QUESTION:
            return{
                ...state,
                questions: state.questions.concat(action.payload.id)
            }

        case SET_POLL_OPTION:

            const {qid, answer} = action;
            return{
                ...state,
                answers: {
                    ...state.answers,
                    [qid]: answer
                }
            }

        default:
            return state;
    }
}