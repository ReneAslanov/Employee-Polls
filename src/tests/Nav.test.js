import Nav from "../components/Nav";
import store from "../store";
import { Provider } from "react-redux";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { setAuthedUser } from "../actions/authedUser";

describe("Nav", () => {
    it("should have all the links present", async() => {
        const testAuthedUser = {
            id: "sarahedo",
            avatarURL: "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcS3_juC_cG7iLmwAaK4Mxxy-OtJ52335hZfHhiS5eSHGp9a0C9OY67183F9TV4t01DNILs&usqp=CAU",
            answers: {
                "8xf0y6ziyjabvozdd253nd": 'optionOne',
                "6ni6ok3ym7mf1p33lnez": 'optionOne',
                "am8ehyc8byjqgar0jgpub9": 'optionTwo',
                "loxhs1bqm25b708cmbf3g": 'optionTwo'
              },
            questions: ['8xf0y6ziyjabvozdd253nd', 'am8ehyc8byjqgar0jgpub9']
        }

        store.dispatch(setAuthedUser(testAuthedUser));

        const view = render(
            <MemoryRouter initialEntries={["/home"]}>
                <Provider store={store}>
                    <Nav />
                </Provider>
            </MemoryRouter>
        );
        expect(view).toMatchSnapshot();

        const home = screen.getByText("Home");
        const leaderboard = screen.getByText("Leaderboard");
        const newLink = screen.getByText("New");
        const logout = screen.getByText("Logout");

        expect(home).toBeInTheDocument();
        expect(leaderboard).toBeInTheDocument();
        expect(newLink).toBeInTheDocument();
        expect(logout).toBeInTheDocument();
    })

})