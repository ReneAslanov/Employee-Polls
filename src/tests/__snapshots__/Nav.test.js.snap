// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should have all the links present 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nav-wrapper"
      >
        <div
          class="nav-routes"
        >
          <ul
            class="nav-list"
          >
            <a
              href="/home"
            >
              Home
            </a>
            <a
              href="/leaderboard"
            >
              Leaderboard
            </a>
            <a
              href="/add"
            >
              New
            </a>
          </ul>
        </div>
        <div
          class="nav-user"
        >
          <ul
            class="nav-list"
          >
            <div
              class="nav-img-wrapper"
            >
              <img
                alt="just an avatar"
                class="nav-avatar"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcS3_juC_cG7iLmwAaK4Mxxy-OtJ52335hZfHhiS5eSHGp9a0C9OY67183F9TV4t01DNILs&usqp=CAU"
              />
            </div>
            <p
              class="nav-userName"
            >
              sarahedo
            </p>
            <a
              href="/"
            >
              Logout
            </a>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nav-wrapper"
    >
      <div
        class="nav-routes"
      >
        <ul
          class="nav-list"
        >
          <a
            href="/home"
          >
            Home
          </a>
          <a
            href="/leaderboard"
          >
            Leaderboard
          </a>
          <a
            href="/add"
          >
            New
          </a>
        </ul>
      </div>
      <div
        class="nav-user"
      >
        <ul
          class="nav-list"
        >
          <div
            class="nav-img-wrapper"
          >
            <img
              alt="just an avatar"
              class="nav-avatar"
              src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcS3_juC_cG7iLmwAaK4Mxxy-OtJ52335hZfHhiS5eSHGp9a0C9OY67183F9TV4t01DNILs&usqp=CAU"
            />
          </div>
          <p
            class="nav-userName"
          >
            sarahedo
          </p>
          <a
            href="/"
          >
            Logout
          </a>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
